### PR TITLE
Service Reset Password URL

### PR DESCRIPTION
**Pull request by guyyoo**

### CODE REVIEW

File: src/rootPages/Designer/properties/process/ABProcessTaskService.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+       *        the webix ui.id container for the properties panel.
+       */
+      switchTo(key) {
+         const ids = this.ids;
+
+         const values = this.values();
+
+         values.id = this.element.id;
+         values.diagramID = this.element.diagramID;
+         values.key = key || this.key;
```
Comment (wongpratan) [inline src/rootPages/Designer/properties/process/ABProcessTaskService.js:128]: Should It be 
`key ?? this.key; `
instead ?

File: src/rootPages/Designer/properties/process/ABProcessTaskService.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+      values() {
+         const ids = this.ids;
+
+         let obj = {};
+
+         const $name = $$(ids.name);
+
+         obj.label = $name.getValue() || "";
+         obj.name = $name.getValue() || "";
```
Comment (wongpratan) [inline src/rootPages/Designer/properties/process/ABProcessTaskService.js:158]: ??

File: src/rootPages/Designer/properties/process/ABProcessTaskService.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      }
+
+      populate(element) {
+         const ids = this.ids;
+
+         const $name = $$(ids.name);
+
+         this.element = element;
+
+         $name.setValue(element.label || "");
```
Comment (wongpratan) [inline src/rootPages/Designer/properties/process/ABProcessTaskService.js:170]: `$name.setValue(element.label ?? "");`

File: src/rootPages/Designer/properties/process/ABProcessTaskServiceGetResetPasswordUrl.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+         const ids = this.ids;
+
+         const $name = $$(ids.name);
+         const $email = $$(ids.email);
+         const $url = $$(ids.url);
+
+         obj.label = $name.getValue() || "";
+         obj.name = $name.getValue() || "";
+         obj.email = $email.getValue() || "";
+         obj.url = $url.getValue() || "";
```
Comment (wongpratan) [inline src/rootPages/Designer/properties/process/ABProcessTaskServiceGetResetPasswordUrl.js:113]: :)

**Review by wongpratan — APPROVED** [review]

File: src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+            });
+
+            repeatObjectFields = this.getFieldOptions(
+               fieldRepeat.datasourceLink
+            );
+         }
+
+         setOptions.push({
+            id: 6,
+            value: `Set by the paremeter of a Query task`,
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js:237]: this should be a `L("Set by the parameter of a Query task")`   <--- also notice spelling of `parameter`

File: src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+         });
+      }
+
+      /**
+       * @method propertiesStash()
+       * pull our values from our property panel.
+       * @param {string} id
+       *        the webix $$(id) of the properties panel area.
+       */
+      propertiesStash() {
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js:326]: our new Property views use `populate()`  and `values()`  now.  refactor this old `propertiesStash()` to be `values()`

Comment (Hiro-Nakamura) [reply src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js:326]: or just get rid of this if it is unnecessary

File: src/rootPages/Designer/properties/PropertyManager.js [hunk 7]
```diff
@@ -34,13 +34,14 @@ export default function (AB) {
    });
 
    var Processes = [];
-   // {array}
-   // All the ABProcess... Property Interfaces Available
    [
       require("./process/ABProcessEnd.js"),
-      require("./process/ABProcessParticipant.js"),
```
Comment (Hiro-Nakamura) [inline src/rootPages/Designer/properties/PropertyManager.js:41]: Do we not need this one anymore?

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]

File: src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+      // {string}
+      // This should match the ABProcessTriggerLifecycleCore.defaults().key value.
+
+      ui() {
+         // we are creating these on the fly, and should have CurrentApplication
+         // defined already.
+
+         const ids = this.ids;
+
+         const objectList = this.AB.objects().map((o) => {
```
[the remark this answers is not in the document]
Comment (guyyoo) [reply src/rootPages/Designer/properties/process/ABProcessTaskServiceInsertRecord.js]: I already talked with him. Yes, we need to see only the objects in the current app we work on. I am going to change this.

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
fix some of those edits in the PropertyManager and then should be good to go.

**Pull request merged**